### PR TITLE
add public-good to default policy

### PR DESCRIPTION
**Pull request by bdehamer**
Adds a second `Authority` to the default clusterimagepolicy which will allow images which have been signed by PGI Sigstore.

This essentially means that the policy would allow ANY image as long as it had been signed by EITHER the GitHub Fulcio OR the PGI Fulcio.

Either the github or sigstore PGI authorities can be selectively disabled with the `trust.github` and `trust.sigstorePublic` values in the `values.yaml` file.

### CODE REVIEW

**Comment by kommendorkapten** [comment]
Could this be parameterized during installation? I'm thinking what the best default values are here. The policy would still only apply to images coming from the configured organization, so adding PGI to the trust root is not a massive change in trust boundary.

File: charts/trust-policies/templates/clusterimagepolicy-github.yaml [hunk 1]
```diff
@@ -18,4 +19,15 @@ spec:
...
+    keyless:
+      identities:
+      - issuer: https://token.actions.githubusercontent.com
+        subjectRegExp: https://github.com/bdehamer/.*/\.github/workflows/.*
+    ctlog:
+      url: https://rekor.sigstore.dev
+    signatureFormat: bundle
+    attestations:
+    - name: require-attestation
+      predicateType: https://slsa.dev/provenance/v1
```
Comment (codysoyland) [inline charts/trust-policies/templates/clusterimagepolicy-github.yaml:32]:
```suggestion
      predicateType: {{ .Values.policy.predicateType }}
```

**Comment by codysoyland** [comment]
> Could this be parameterized during installation? I'm thinking what the best default values are here. The policy would still only apply to images coming from the configured organization, so adding PGI to the trust root is not a massive change in trust boundary.

Yeah, this is a good question. I don't think we should default to trusting both PGI and GitHub. I've done testing on an arrangement similar to this PR and there are pros/cons.

On the positive side, this is simple and supports both trust boundaries, and as @kommendorkapten said, it's not a massive change in trust boundary. This option does not require any custom configuration to support both PGI and GitHub.

On the negative side, this does introduce additional latency during verification, as both trust roots are attempted during every invocation. The PGI verifier will always try to hit Rekor during verification, which both adds latency, incurs load on PGI Rekor, and creates a privacy issue due to the Rekor request.

Additionally, if we add new trust boundaries (as has been discussed in private), we will want a way to configure them on/off.

I do think the most common use case is private repos (GitHub trust root), so I would prefer that to remain the default.

I would add Helm values to the effect of:

```yaml
trust:
  github: true
  sigstore-public: false
```

**Comment by bdehamer** [comment]
## Default
```
helm template --debug ./charts/trust-policies \
  --set policy.enabled=true \
  --set policy.organization=FOO 
```

```yaml
apiVersion: policy.sigstore.dev/v1alpha1
kind: ClusterImagePolicy
metadata:
  name: github-policy
spec:
  images:
  - glob: "**"
  authorities:

  - name: github
    keyless:
      trustRootRef: github
      identities:
      - issuer: https://token.actions.githubusercontent.com
        subjectRegExp: https://github.com/FOO/.*/\.github/workflows/.*
    rfc3161timestamp:
      trustRootRef: github
    signatureFormat: bundle
    attestations:
    - name: require-attestation
      predicateType: https://slsa.dev/provenance/v1

  - name: public-good
    keyless:
      identities:
      - issuer: https://token.actions.githubusercontent.com
        subjectRegExp: https://github.com/FOO/.*/\.github/workflows/.*
    ctlog:
      url: https://rekor.sigstore.dev
    signatureFormat: bundle
    attestations:
    - name: require-attestation
      predicateType: https://slsa.dev/provenance/v1
```
## Disable GitHub
```
helm template --debug ./charts/trust-policies \
  --set policy.enabled=true \
  --set trust.github=false \
  --set policy.organization=FOO 
```

```yaml
apiVersion: policy.sigstore.dev/v1alpha1
kind: ClusterImagePolicy
metadata:
  name: github-policy
spec:
  images:
  - glob: "**"
  authorities:

  - name: public-good
    keyless:
      identities:
      - issuer: https://token.actions.githubusercontent.com
        subjectRegExp: https://github.com/FOO/.*/\.github/workflows/.*
    ctlog:
      url: https://rekor.sigstore.dev
    signatureFormat: bundle
    attestations:
    - name: require-attestation
      predicateType: https://slsa.dev/provenance/v1
```

## Disable Sigstore Public Good
```
helm template --debug ./charts/trust-policies \
  --set policy.enabled=true \
  --set trust.sigstorePublic=false \
  --set policy.organization=FOO 
```

```yaml
apiVersion: policy.sigstore.dev/v1alpha1
kind: ClusterImagePolicy
metadata:
  name: github-policy
spec:
  images:
  - glob: "**"
  authorities:

  - name: github
    keyless:
      trustRootRef: github
      identities:
      - issuer: https://token.actions.githubusercontent.com
        subjectRegExp: https://github.com/FOO/.*/\.github/workflows/.*
    rfc3161timestamp:
      trustRootRef: github
    signatureFormat: bundle
    attestations:
    - name: require-attestation
      predicateType: https://slsa.dev/provenance/v1
```

**Comment by codysoyland** [comment]
Thanks @bdehamer! Now that I'm seeing the policies and examples, I'm wondering if the `trust` value should be nested under `policy`, so for example the command to disable github prod would be:

```
helm template --debug ./charts/trust-policies \
  --set policy.enabled=true \
  --set policy.trust.github=false \
  --set policy.organization=FOO 
```

What do you think about that?

**Review by codysoyland — APPROVED** [review]

**Pull request merged**